File: solutions/javascript/line-up/1/line-up.js
//
// This is only a SKELETON file for the 'Line Up' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const format = (name, number) => {
  let result = number + "th";
  switch (number.toString()[number.toString().length - 1]) {
    case "2": number != 12 && number.toString()[number.toString().length - 2] != 1 ? result = number + "nd" : "12th";
      break;
    case "3": number != 13 && number.toString()[number.toString().length - 2] != 1 ? result = number + "rd" : "13th";
      break;
    case "1": number != 11 && number.toString()[number.toString().length - 2] != 1 ? result = number + "st" : "11th";
      break;
    }
  return name + ", you are the " + result + " customer we serve today. Thank you!";
};

File: solutions/csharp/line-up/1/LineUp.cs
public static class LineUp
{
    public static string Format(string name, int number)
    {
        if (number.ToString().EndsWith("1") && !number.ToString().EndsWith("11"))
            return $"{name}, you are the {number}st customer we serve today. Thank you!";
        if (number.ToString().EndsWith("2") && !number.ToString().EndsWith("12"))
            return $"{name}, you are the {number}nd customer we serve today. Thank you!";
        if (number.ToString().EndsWith("3") && !number.ToString().EndsWith("13"))
            return $"{name}, you are the {number}rd customer we serve today. Thank you!";
        return $"{name}, you are the {number}th customer we serve today. Thank you!";
    }
}
